snuba/web/rpc/common/exceptions.py

```diff
@@ -0,0 +1,31 @@
+from typing import Union
+
+from sentry_protos.snuba.v1.error_pb2 import Error as ErrorProto
+
+from snuba.web import QueryException
+
+
+class RPCRequestException(Exception):
+    status_code: int
+
+    def __init__(self, status_code: int, message: str):
+        self.status_code = status_code
+        super().__init__(message)
+
+
+class BadSnubaRPCRequestException(RPCRequestException):
```

Comment:
Bad/Good are too general. Since it's a 400 error by default it means it's a user error. Let's communicate that
```suggestion
class InvalidSnubaRPCRequestException(RPCRequestException):
```

Reply:
400 is called 'badrequest' semantically

Reply:
fair nuff